### ISSUE TITLE
Latest version from Jenkinsapi breaks connection

### ISSUE THREAD

**Issue by jonazpiazu** [ros-infrastructure/ros_buildfarm#597]
When trying to make a clean install, I got this error when running the `generate_all_jobs.py` script (using Jenkins version 2.89.4):

```
rosin-test-alpha: Connecting to Jenkins 'http://master:8080'
    rosin-test-alpha: ERROR:root:Failed request at http://master:8080/api/python with params: {'tree': 'jobs[name,color,url]'} jobs[name,color,url]
    rosin-test-alpha: Traceback (most recent call last):
    rosin-test-alpha:   File "/root/venv/bin/generate_all_jobs.py", line 321, in <module>
    rosin-test-alpha:     main()
    rosin-test-alpha:   File "/root/venv/bin/generate_all_jobs.py", line 70, in main
    rosin-test-alpha:     jenkins = connect(config.jenkins_url)
    rosin-test-alpha:   File "/root/venv/lib/python3.5/site-packages/ros_buildfarm/jenkins.py", line 63, in connect
    rosin-test-alpha:     jenkins = JenkinsProxy(jenkins_url, username=username, password=password)
    rosin-test-alpha:   File "/root/venv/lib/python3.5/site-packages/ros_buildfarm/jenkins.py", line 42, in __init__
    rosin-test-alpha:     super(JenkinsProxy, self).__init__(*args, **kwargs)
    rosin-test-alpha:   File "/root/venv/lib/python3.5/site-packages/jenkinsapi/jenkins.py", line 63, in __init__
    rosin-test-alpha:     JenkinsBase.__init__(self, baseurl, poll=not lazy)
    rosin-test-alpha:   File "/root/venv/lib/python3.5/site-packages/jenkinsapi/jenkinsbase.py", line 35, in __init__
    rosin-test-alpha:     self.poll()
    rosin-test-alpha:   File "/root/venv/lib/python3.5/site-packages/jenkinsapi/jenkinsbase.py", line 57, in poll
    rosin-test-alpha:     data = self._poll(tree=tree)
    rosin-test-alpha:   File "/root/venv/lib/python3.5/site-packages/jenkinsapi/jenkins.py", line 68, in _poll
    rosin-test-alpha:     if not tree else tree)
    rosin-test-alpha:   File "/root/venv/lib/python3.5/site-packages/jenkinsapi/jenkinsbase.py", line 81, in get_data
    rosin-test-alpha:     response.raise_for_status()
    rosin-test-alpha:   File "/root/venv/lib/python3.5/site-packages/requests/models.py", line 940, in raise_for_status
    rosin-test-alpha:     raise HTTPError(http_error_msg, response=self)
    rosin-test-alpha: requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: http://master:8080/api/python?tree=jobs%5Bname%2Ccolor%2Curl%5D
```

I checked the url, username and password and they are correct.

Looking at the jenkinsapi [release history](https://pypi.org/project/jenkinsapi/#history), there was a new version released recently. I have tried to manually install the previous version (v. 0.3.6) and this one does  not show the problems above, so it looks like the newest version broke something.

**Comment by jonazpiazu** [comment]
Btw, it seems to be related with the connection parameters provided, as using the newest jenkinsapi 0.3.7 in a Python console connects correctly:

```Python
In [1]: from jenkinsapi.jenkins import Jenkins 
   ...: jenkins = Jenkins('http://master:8080', 'admin', 'changeme') 
   ...: print(jenkins.version)                                                                       
2.89.4
```

**Comment by dirk-thomas** [comment]
This looks like a bug reported and fixed upstream (pycontribs/jenkinsapi#679) which was present in version 0.3.7 `jenkinsapi` and has been fixed in version 0.3.8.

I was able to reproduce the problem with 0.3.7 and it works for me with 0.3.8. Therefore I will go ahead and close this ticket. Please feel free to comment if the problem isn't resolved for you by updating `jenkinsapi`.
